### ISSUE TITLE
Over LTE (Cellular) Video Streaming Optimization

### ISSUE THREAD

**Issue by fpuigb** [hjdhjd/homebridge-unifi-protect2#48]
I have been playing with the plug-in and the video streaming over Wi-Fi is very smooth with a loading time of few seconds. When I try it over LTE or another Wi-Fi the video streaming experience its not than smooth. The loading time is around 10 seconds when load, other times shows unavailable. Over Wi-Fi the streaming is stablished @ 1280x720, 30 fps, 299 kbps. over LTE is stablished @ 640x360, 30 fps, 132 kbps. Is there a way to optimize the over cellular experience reducing fps or kbps? Also, I tried over another Wi-Fi connection and the experience is the same as the cellular one. Not sure where the issue is... The internet connection upload is 300 Mbps with very low latency and the LTE connection is around 180 Mbps. The Protect App works perfect over LTE. Any idea?

**Comment by hjdhjd** [comment]
HomeKit directly controls the rates. You (or I) have no control over it. Everything you've described is well within the norms of how HomeKit works. HomeKit decides what it wants to stream or where.

What you're showing above is what **HomeKit** is requesting, not what the plugin is choosing on it's own.

As to latency...your latency at home is really a non-issue. It's the latency of LTE, and it's always substantial, and it adds up. The native Protect app is able to optimize the experience because they control all aspects of the communication back...meaning, they *know* they're on cellular and can choose the best option to deal with it.

I don't have that luxury in HomeKit because Apple controls those portions of the experience and I can't really do much other than give it what it's requesting. Ideally, Apple will continue to optimize the non-WiFi experience.

Best of luck.

TL;DR

**Comment by fpuigb** [comment]
Thanks for the info. BTW; Great plug-in...

**Comment by hjdhjd** [comment]
Your welcome, and appreciate the feedback. Good luck!
